recipes/confu_json/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+from conans import ConanFile, tools
+from conans.tools import check_min_cppstd
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+from conans.tools import Version
+
+
+class ConfuJson(ConanFile):
+    name = "confu_json"
+    homepage = "https://github.com/werto87/confu_json"
+    description = "uses boost::fusion to help with serialization; json <-> user defined type"
+    topics = ("json parse", "serialization", "user defined type")
+    license = "BSL-1.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 20
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "10",
+            "clang": "10",
+        }
+
+    def configure(self):
+        if self.settings.compiler == "apple-clang":
+            raise ConanInvalidConfiguration(
+                "apple-clang is not supported because of missing concept support")
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} "
+                             "compiler support.".format(
+                                 self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(
+                    "{} requires C++{} support. "
+                    "The current compiler {} {} does not support it.".format(
+                        self.name, self._minimum_cpp_standard,
+                        self.settings.compiler,
+                        self.settings.compiler.version))
+        self.options["boost"].header_only = True
```

Comment:
Let's not surprises the consumer 

```suggestion

    def configure(self):
        if not self.options["boost"].header_only:
            raise ConanInvalidConfiguration("{} requires a header-only Boost. ".format(self.name))
```